### PR TITLE
New map messages produces wargame with wrong id #1656

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

**Comment by IanMayo** [comment]
Thanks @lilitkarapetyan .

But, once the wargame has been initiated, we should create a new document with the `_id` of `settings`.

So, before initiated:
- ***overwrite*** document named `initial_wargame`

After initiated:
- ***add new version*** of document named `settings`

File: packages/client/src/api/wargames_api/index.ts [hunk 1]
```diff
@@ -332,7 +332,7 @@ export const initiateGame = (dbName: string): Promise<MessageInfoType> => {
     const messageInfoType: MessageInfoType = {
       ...wargame,
       _rev: undefined,
-      _id: dbDefaultSettings._id,
+      _id: new Date().toISOString(),
```
Comment (IanMayo) [inline packages/client/src/api/wargames_api/index.ts]: Hmm, @lilitkarapetyan  - so we don't use an id of `settings` for all versions of the wargame after it has been initiated?

**Comment by IanMayo** [comment]
Ok. It's a start, but something still broken.

1. Login as Game Control
2. Do "umpire laydown" (move 4 assets)
3. Submit "umpire laydown"
4. Go to Game Admin page
5. Click on `wargame initiated`
6. Log in as `Red CO`
7. Note we're now in `Force laydown`
8. Do force laydown for `MPA 2`. Remember where MPA2 asset placed
9. Submit force laydown
10. Notice where `MPA 2` is
11. Refresh page
12. `MPA 2` back in `Force laydown`
13. Look in database:
image

The results of `Force laydown` are in document titled `initial_wargame`.  But, this document isn't supplied to the client on refresh.  It _may_ be filtered by this code:
image

So, `get all documents for wargame` doesn't return the one titled `initial_wargame`.  This is the one with the results of Red `force laydown` (i.e. a position for `MPA 2`).

**Comment by IanMayo** [comment]
Thanks for the update @lilitkarapetyan , but isn't `settings` the name we use for the wargame document once it has been initiated?

**Comment by IanMayo** [comment]
Sorry, @lilitkarapetyan - this is still broken.
image

I did `umpire-laydown`, then `initiate wargame`.  This produced a wargame with the new id (`initial_wargame_start`). But, when I changed a channel name & did `Save channel`,  it created a wargame with a date-time id.
image

**Comment by IanMayo** [comment]
Ok, I've made some progress with this today.  I'll carry on tomorrow. I will be doing more refactoring of `wargames_api`. I haven't really worked with it before, but there are some bits that look wrong.

**Comment by lilitkarapetyan** [comment]
> 



> Sorry, @lilitkarapetyan - this is still broken. image
> 
> I did `umpire-laydown`, then `initiate wargame`. This produced a wargame with the new id (`initial_wargame_start`). But, when I changed a channel name & did `Save channel`, it created a wargame with a date-time id. image

Hi Ian, I will check

**Comment by lilitkarapetyan** [comment]
> Ok, I've made some progress with this today. I'll carry on tomorrow. I will be doing more refactoring of `wargames_api`. I haven't really worked with it before, but there are some bits that look wrong.

Ok, i'll look into it when you're done

**Comment by IanMayo** [comment]
Closing this PR, since Ian mangled the code. Superceded by #1668

**Pull request closed without merging**